flake8_trio.py

```diff
@@ -1421,9 +1448,89 @@ def visit_AsyncFunctionDef(
 
     def visit_Call(self, node: ast.Call):
         if self.async_function and not getattr(node, "awaited", False):
-            blocking_calls = self.options.trio200_blocking_calls
-            if key := fnmatch_qualified_name([node.func], *blocking_calls):
-                self.error(node, key, blocking_calls[key])
+            self.visit_blocking_call(node)
+
+    def visit_blocking_call(self, node: ast.Call):
+        blocking_calls = self.options.trio200_blocking_calls
+        if key := fnmatch_qualified_name([node.func], *blocking_calls):
+            self.error(node, key, blocking_calls[key])
+
+
+@error_class
+class Visitor21X(Visitor200):
+    error_codes = {
+        "TRIO210": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO211": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO212": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+    }
+
+    def visit_blocking_call(self, node: ast.Call):
+        http_methods = (
+            "get",
+            "options",
+            "head",
+            "post",
+            "put",
+            "patch",
+            "delete",
+            "request",
+            "send",
+        )
+        for http_package in "requests", "httpx":
+            if get_matching_call(node, *http_methods, base=http_package):
+                self.error(node, ast.unparse(node.func), error_code="TRIO210")
+                return
+
+        func_name = ast.unparse(node.func)
+        if func_name == "urllib3.requests":
+            self.error(node, func_name, error_code="TRIO211")
+        elif func_name in ("urllib.request.urlopen", "request.urlopen", "urlopen"):
+            self.error(node, func_name, error_code="TRIO212")
+
+
+# Process invocations 202
+@error_class
+class Visitor22X(Visitor200):
+    error_codes = {
+        "TRIO220": (
+            "Sync call {} in async function, use "
+            "`await nursery.start(trio.run_process, ...)`"
+        ),
+        "TRIO221": "Sync call {} in async function, use `await trio.run_process(...)`",
+    }
+
+    def visit_blocking_call(self, node: ast.Call):
+        def is_p_wait(arg: ast.expr) -> bool:
+            return (isinstance(arg, ast.Attribute) and arg.attr == "P_WAIT") or (
+                isinstance(arg, ast.Name) and arg.id == "P_WAIT"
+            )
+
+        func_name = ast.unparse(node.func)
+        if func_name in ("subprocess.Popen", "os.popen"):
+            self.error(node, func_name, error_code="TRIO220")
+            return
+
+        if func_name in "os.system" or func_name.startswith("subprocess."):
```

Comment:
Not if we construct a `subprocess.CompletedProcess` object, or a `subprocess.CalledProcessError`, etc.  Needs to be an explicit list, sadly.